.github/workflows/ci.yml

```diff
@@ -87,7 +87,7 @@ jobs:
 
   docs-snippets:
     name: docs-snippets
-    needs: [docs-changes, ci-is-on-main-repo]
+    needs: [unit-tests, docs-changes, ci-is-on-main-repo]
```

Comment:
If unit-tests are failing then we shouldn't bother running the doc-snippet tests.